accepted/future-releases/enhanced-enums/feature-specification.md

```diff
@@ -68,128 +66,175 @@ must redirect to a generative constructor._
 
 ## Semantics
 
-First we will add a `const Enum();` constructor to the `Enum` class, so that it can validly be a superclass of `enum` classes. _(The `Enum` class currently has a non-`const` default constructor that no-one can use because the class is abstract and cannot be extended, and in practice `enum` classes implement it instead. We now need to, at least pretend to, extend it so that mixins `on Enum` can be applied.)_
+We change the publicly visible API of the  `Enum` class to be declared as:
 
-The semantics of such an enum declaration is defined as being equivalent to *rewriting into a corresponding class declaration* as follows:
-
-- Declare a class with the same name and type parameters as the `enum` declaration.
-
-- Add `extends Enum`, where `Enum` refers to the type declared in `dart:core`.
+```dart
+class Enum {
+  // No default constructor (however we enforce that).
+  external int get index;
+  external String toString();
+}
+```
 
-- Further add the mixins and interfaces of the `enum` declaration.
+We intend to (at least pretend to) let `enum` classes extend `Enum`, and let mixins and members access the default `index` and `toString()` through `super.`. _(In practice, we may use a different class implementing `Enum` as the superclass, but for checking the validity of `super.index`/`super.toString()`, we analyze against `Enum` itself, so it must have non-abstract implementations.)_
+
+This all makes it look as if `Enum` would be a valid superclass for the mixin applications and methods of the enhanced `enum` class.
+
+The semantics of such an enum declaration, *E*, is defined as introducing a (semantic) *class*, *C*, just like a similar `class` declaration.
+
+* **Name**: The name of the class *C* and its implicit interface is the name of the `enum` declaration.
+
+* **Superclass**: The superclass of *C* is (seemingly) the `Enum` class, with the mixins declared by *E* applied. _(In practice, the superclass may be another system provided class with implements `Enum`.)_
```

Comment:
Can we say something like this?: "The super class of *C* is a built-in class _EnumImpl_. _EnumImpl_ may be `Enum`, or it may be a different class, but the interface of _EnumImpl_ as seen from a non-system library is the same as that of `Enum`, and _EnumImpl_ is a subtype of `Enum`."

Reply:
Something like that.